tests/src/JIT/opt/ObjectStackAllocation/ObjectStackAllocationTests.cs

```diff
@@ -0,0 +1,232 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Reflection;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace ObjectStackAllocation
+{
+    class SimpleClassA
+    {
+        public int f1;
+        public int f2;
+
+        public SimpleClassA(int f1, int f2)
+        {
+            this.f1 = f1;
+            this.f2 = f2;
+        }
+    }
+
+    sealed class SimpleClassB
+    {
+        public long f1;
+        public long f2;
+
+        public SimpleClassB(long f1, long f2)
+        {
+            this.f1 = f1;
+            this.f2 = f2;
+        }
+    }
+
+    class SimpleClassWithGCField : SimpleClassA
+    {
+        public object o;
+
+        public SimpleClassWithGCField(int f1, int f2, object o) : base(f1, f2)
+        {
+            this.o = o;
+        }
+    }
+
+    class ClassWithNestedStruct
+    {
+        public ClassWithNestedStruct(int f1, int f2)
+        {
+            ns.f1 = f1;
+            ns.f2 = f2;
+            ns.s.f1 = f1;
+            ns.s.f2 = f2;
+        }
+
+        public NestedStruct ns;
+    }
+
+    struct SimpleStruct
+    {
+        public int f1;
+        public int f2;
+    }
+
+    struct NestedStruct
+    {
+        public int f1;
+        public int f2;
+        public SimpleStruct s;
+    }
+
+    class Tests
+    {
+        static volatile int f1 = 5;
+        static volatile int f2 = 7;
+
+        delegate int Test();
+
+        static int methodResult = 100;
+
+        public static int Main()
+        {
+            bool spcOptimizationEnabled = SPCOptimizationsEnabled();
+
+            CallTestAndVerifyAllocation(AllocateSimpleClassAndAddFields, 12, !spcOptimizationEnabled);
```

Comment:
These allocation amounts are in bytes? If so they may vary by architecture...

Reply:
No, these are return values from the methods. I'm not checking the exact allocation bytes, just whether heap allocation amount changed.